src/modules/notifications/notificaitons.ts

```diff
@@ -0,0 +1,38 @@
+/* eslint-disable @typescript-eslint/indent */
+import Mailgun from 'mailgun-js';
+import Email from '../../lib/emails';
+
+/**
+ * @description Send out email notificaions from the subList
+ * @param {Array<string>} subList A list of subscribers
+ * @param {Date} deliveryTime Time that the email will be delivered
+ */
+const notifyMany = async (
+    subList: Array<string>,
+    deliveryTime: Date
+): Promise<Array<string | Mailgun.messages.SendResponse>> => {
+    if (subList.length === 0) {
+        throw new Error('Empty subscribe list');
+    }
+    const results: Array<Promise<string | Mailgun.messages.SendResponse>> = [];
+    while (subList.length) {
+        // Take max of 1k emails to send for each mailgun batch
+        const emails = subList.splice(0, 1000);
+        // TODO Figure out what should be sent for notification
+        const recipientVariables = '';
+        const notificationString = 'Test Notification';
+        // eslint-disable-next-line no-await-in-loop
+        results.push(
+            Email.sendEmail(
+                emails,
+                'Prytaneum Notification',
+                notificationString,
+                deliveryTime,
+                recipientVariables
+            )
+        );
+    }
```

Comment:
I think a better approach here would be to use the `.slice()` method.

With how it's currently coded, the while loop depends on the mutability of the length property of the subList array, which with javascript isn't always the best idea unless necessary.  In future versions of the program, if someone working on it were a bit careless, then it's possible to create an infinite loop here (say they add a condition, change how you're currently splicing, etc.).  

With `.slice()` you can just use a plain old for-loop and no matter what there's no possibility (almost) for future programmers to make an infinite loop mistake.

Granted, the tests should catch if this is a problem (although tests only timeout after x amount of time with an infinite loop, so it may not be obvious to track down at first), but lets just reduce the number of ways people (or ourselves) could mess up our code in the future.